diff --git a/bin/stty/cchar.c b/bin/stty/cchar.c
index ad88c76..fc7bc83 100644
--- a/bin/stty/cchar.c
+++ b/bin/stty/cchar.c
@@ -63,6 +63,7 @@ const struct cchar cchars1[] = {
 	{ "eol",	VEOL,		CEOL },
 	{ "eol2",	VEOL2,		CEOL },
 	{ "erase",	VERASE,		CERASE },
+	{ "worderase",	VWORDERASE,	CWORDERASE },
 	{ "intr",	VINTR,		CINTR },
 	{ "kill",	VKILL,		CKILL },
 	{ "lnext",	VLNEXT,		CLNEXT },
diff --git a/lib/libedit/tty.h b/lib/libedit/tty.h
index 7bdb0b2..ead0010 100644
--- a/lib/libedit/tty.h
+++ b/lib/libedit/tty.h
@@ -172,6 +172,9 @@
 #ifndef CERASE
 # define CERASE		0177	/* ^? */
 #endif /* CERASE */
+#ifndef CWORDERASE
+# define CWORDERASE	CONTROL('l')
+#endif /* CWORDERASE */
 #ifndef CKILL
 # define CKILL		CONTROL('u')
 #endif /* CKILL */
diff --git a/sys/sys/termios.h b/sys/sys/termios.h
index d1e6d79..3138484 100644
--- a/sys/sys/termios.h
+++ b/sys/sys/termios.h
@@ -76,7 +76,8 @@
 #define VSTATUS		18	/* ICANON */
 /*			19	   spare 2 */
 #endif
-#define	NCCS		20
+#define        VWORDERASE	20	/* ICANON */
+#define	NCCS		21
 
 #define _POSIX_VDISABLE	__CAST(unsigned char, '\377')
 
diff --git a/sys/sys/ttydefaults.h b/sys/sys/ttydefaults.h
index 42c6758..910eef6 100644
--- a/sys/sys/ttydefaults.h
+++ b/sys/sys/ttydefaults.h
@@ -63,8 +63,10 @@
 #define	CEOL		((unsigned char)'\377')	/* XXX avoid _POSIX_VDISABLE */
 #if defined(__minix)
 #define	CERASE		CTRL('h')
+#define CWORDERASE	CTRL('l')
 #else
 #define	CERASE		0177
+#define CWORDERASE	0154
 #endif /* defined(__minix) */
 #define	CINTR		CTRL('c')
 #define	CSTATUS		CTRL('t')
@@ -99,6 +101,7 @@ const cc_t ttydefchars[NCCS] = {
 	[VEOL] = CEOL,
 	[VEOL2] = CEOL,
 	[VERASE] = CERASE,
+	[VWORDERASE] = CWORDERASE,
 	[VWERASE] = CWERASE,
 	[VKILL] = CKILL,
 	[VREPRINT] = CREPRINT,
