/**
 * @author Fabian Klopfer
 * @author Jannik Bamberger
 * @credits 13
 */

diff --git a/bin/date/date.c b/bin/date/date.c
index 4a11d02..962329d 100644
--- a/bin/date/date.c
+++ b/bin/date/date.c
@@ -42,6 +42,7 @@ static char sccsid[] = "@(#)date.c	8.2 (Berkeley) 4/28/95";
 #else
 __RCSID("$NetBSD: date.c,v 1.60 2011/08/27 12:55:09 joerg Exp $");
 #endif
+
 #endif /* not lint */
 
 #include <sys/param.h>
@@ -75,6 +76,7 @@ __dead static void usage(void);
 int
 main(int argc, char *argv[])
 {
+	char kalenderwoche[80];
 	char *buf;
 	size_t bufsiz;
 	const char *format;
@@ -157,6 +159,17 @@ badarg:				 errx(EXIT_FAILURE,
 			goto bad;
 
 	(void)printf("%s\n", buf + 1);
+
+	strftime(kalenderwoche,80,"%W", tm);	
+
+	(void)printf("%s%s\n", "Kalenderwoche: ", kalenderwoche);
+	
+	srand(time(NULL));
+
+	if(rand() % 42 == 1) {
+		(void)printf("%s\n", "25.08.1991"); 
    // Die Formatierung ist nicht wie beim Standardverhalten (-1)
    // Dieses Datum wird zusätzlich ausgegeben (-1)
+	}	
+	
 	free(buf);
 	return 0;
 bad:
